platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -714,7 +710,7 @@ public ResolverResults call(BuildOperationContext context) {
                 // TODO: Currently afterResolve runs if there is not an non-unresolved-dependency failure
                 //       We should either _always_ run afterResolve, or only run it if _no_ failure occurred
                 if (!results.getVisitedGraph().getResolutionFailure().isPresent()) {
-                    dependencyResolutionListeners.getSource().afterResolve(incoming);
```

Comment:
❓ Why did this change to a call?

Reply:
And line 822 and 1119

Reply:
These were from another set of changes that this PR was split off from. They are not necessary for this change so I'll revert them. 

But for context, I was trying to deprecate incoming/outgoing so they are only used for the proper roles and I added the role guards to the `getIncoming`/`getOutgoing` methods